### PR TITLE
Enable bf16 support for ck-based convolution solvers in MIOpen

### PR DESCRIPTION
**Pull request by iq136boy**
Enable bf16 support for ck-based convolution solvers in MIOpen

### CODE REVIEW

**Review by junliume — APPROVED** [review]

**Review by amberhassaan — CHANGES_REQUESTED** [review]
@iq136boy : please enable tests for the `bfloat16` by adding a line to each of `test/gtest/group_conv{2,3}d_*.cpp`.

File: test/gtest/group_conv3d_wrw.cpp [hunk 1]
```diff
@@ -35,3 +35,4 @@ DEFINE_GROUP_CONV3D_TEST(float, BackwardWeights);
 // DEFINE_GROUP_CONV3D_TEST(half, BackwardWeights);
 /// \todo int8_t tests don't work. Need debugging
 // DEFINE_GROUP_CONV3D_TEST(int8_t, BackwardWeights);
+DEFINE_GROUP_CONV3D_TEST(bfloat16, BackwardWeights);
```
Comment (amberhassaan) [inline test/gtest/group_conv3d_wrw.cpp:38]: This looks good. Please try uncommenting `half` for backward weights to see if it's still broken.

File: test/gtest/group_conv2d_wrw.cpp [hunk 2]
```diff
@@ -35,3 +35,4 @@ DEFINE_GROUP_CONV2D_TEST(float, BackwardWeights);
 // DEFINE_GROUP_CONV2D_TEST(half, BackwardWeights);
```
Comment (amberhassaan) [inline test/gtest/group_conv2d_wrw.cpp:35]: This looks good. Please try uncommenting `half` for backward weights to see if it's still broken.

**Review by amberhassaan — APPROVED** [review]

File: test/gtest/group_conv2d_wrw.cpp (hunk 2 above)
Comment (iq136boy) [reply test/gtest/group_conv2d_wrw.cpp:35]: > This looks good. Please try uncommenting `half` for backward weights to see if it's still broken.

Hi Amber, sure. I can try to uncomments them to see if they can pass the CI.
Also, I observed some failed test cases in bwd solvers after enable the bf16 supports. Debugging now.

**Comment by junliume** [comment]
@iq136boy the tests starts to fail in CI, due to `Error beyond tolerance Error`
`GroupConv2D_BackwardData_bfloat16_Suite/GroupConv2D_BackwardData_bfloat16.GroupConv2D_BackwardData_bfloat16_Test`

**Comment by iq136boy** [comment]
> @iq136boy the tests starts to fail in CI, due to `Error beyond tolerance Error` `GroupConv2D_BackwardData_bfloat16_Suite/GroupConv2D_BackwardData_bfloat16.GroupConv2D_BackwardData_bfloat16_Test`

@junliume Yes. The error of the kernel output is bigger than the threshold that we set for bwd solver. Most likely, the called ck kernels need to provide more accuracy result unless we can increase the threshold in the gtest.

**Pull request merged**